etc/make_small_lib.pl

```diff
@@ -0,0 +1,335 @@
+      my $common = 0;
+      $common++ while ($last->[$common] && $current->[$common] && $last->[$common] eq $current->[$common]);
+      my $back = @$last - $common;
+      if ($back > 0) {
+        $files .= ("\t" x --$depth) . "</Filter>\r\n" for (1..$back);
+      }
+      my $fwd = [ @$current ]; splice(@$fwd, 0, $common);
+      for my $i (0..scalar(@$fwd) - 1) {
+        $files .= ("\t" x $depth) . "<Filter\r\n";
+        $files .= ("\t" x $depth) . "\tName=\"$fwd->[$i]\"\r\n";
+        $files .= ("\t" x $depth) . "\t>\r\n";
+        $depth++;
+      }
+      $last = $current;
+    }
+    $files .= ("\t" x $depth) . "<File\r\n";
+    $files .= ("\t" x $depth) . "\tRelativePath=\"$full\"\r\n";
+    $files .= ("\t" x $depth) . "\t>\r\n";
+    if ($full =~ $exclude_re) {
+      for (@$targets) {
+        $files .= ("\t" x $depth) . "\t<FileConfiguration\r\n";
+        $files .= ("\t" x $depth) . "\t\tName=\"$_\"\r\n";
+        $files .= ("\t" x $depth) . "\t\tExcludedFromBuild=\"true\"\r\n";
+        $files .= ("\t" x $depth) . "\t\t>\r\n";
+        $files .= ("\t" x $depth) . "\t\t<Tool\r\n";
+        $files .= ("\t" x $depth) . "\t\t\tName=\"VCCLCompilerTool\"\r\n";
+        $files .= ("\t" x $depth) . "\t\t\tAdditionalIncludeDirectories=\"\"\r\n";
+        $files .= ("\t" x $depth) . "\t\t\tPreprocessorDefinitions=\"\"\r\n";
+        $files .= ("\t" x $depth) . "\t\t/>\r\n";
+        $files .= ("\t" x $depth) . "\t</FileConfiguration>\r\n";
+      }
+    }
+    $files .= ("\t" x $depth) . "</File>\r\n";
+  }
+  $files .= ("\t" x --$depth) . "</Filter>\r\n" for (@$last);
+  $files .= "\t</Files>";
+  return $files;
+}
+
+sub process_makefiles {
+  my ($path, @o) = @_;
+  my @all = map {$path.$sep . $_} @o;
+  my $var_o = prepare_makefile_variable("OBJECTS", @o);
+  (my $var_obj = $var_o) =~ s/\.o\b/.obj/sg;
+
+
+  # update MSVC project files
+  s/\.o/.c/ for @all;
+  my $msvc_files = prepare_msvc_files_xml(\@all, qr/NOT_USED_HERE/, ['Debug|Win32', 'Release|Win32', 'Debug|x64', 'Release|x64']);
+  for my $m (qw/libtommath_VS2008.vcproj/) {
+    my $old = read_file($path.$sep.$m);
+    my $new = $old;
+    $new =~ s|<Files>.*</Files>|$msvc_files|s;
+    if ($old ne $new) {
+      write_file($path.$sep.$m, $new);
+      warn "changed: $path$sep$m\n";
+    }
+  }
+
+  # update OBJECTS
+  for my $m (qw/ makefile makefile.shared makefile_include.mk makefile.msvc makefile.unix makefile.mingw /) {
+    my $old = read_file($path.$sep.$m);
+    my $new = $m eq 'makefile.msvc' ? patch_makefile($old, $var_obj)
+                                    : patch_makefile($old, $var_o);
+    if ($old ne $new) {
+      write_file($path.$sep.$m, $new);
+      warn "changed: $path$sep$m\n";
+    }
+  }
+}
+
+# TODO: that is not the best way to do it, but a simple one.
+sub write_header
+{
+  my ($path, @entries) = @_;
+  my $tcpath;
+  my $content = "/* LibTomMath, multiple-precision integer library -- Tom St Denis */\n";
```

Comment:
Here you are just writing a flat list right? I would suggest we remove tommath_class and tommath_superclass.h and replace it with tommath_config.h just containing the flat list. This would be preferable to having multiple different mechanisms around.

Reply:
> Here you are just writing a flat list right?

Yes?

> I would suggest we remove tommath_class and tommath_superclass.h and replace it with tommath_config.h just containing the flat list.

There might be users who already have some kind of automatisms installed. I would like to avoid such brutality just for the sake of prettiness.

Reply:
> There might be users who already have some kind of automatisms installed.

IMO that's the way we should go. Why not create a fresh `tommath_superclass.h` from this tool?